src/token.rs

```diff
@@ -25,7 +26,16 @@ impl std::convert::From<TokenGenerationError> for Rejection {
 }
 
 pub trait TokenGenerator: Clone + Send + Sync {
-    fn generate_token(&mut self) -> Result<String, TokenGenerationError>;
+    fn generate_token(&self) -> Result<String, TokenGenerationError>;
+}
+
+impl<T> TokenGenerator for Arc<T>
+where
+    T: TokenGenerator,
```

Comment:
ooh neat so this is implementing TokenGenerator for Arc\<TokenGenerator\> basically?